source/extensions/transport_sockets/alts/tsi_socket.cc

```diff
@@ -193,14 +260,7 @@ Network::IoResult TsiSocket::doRead(Buffer::Instance& buffer) {
 
   if (handshake_complete_) {
     ASSERT(frame_protector_);
-
-    uint64_t read_size = raw_read_buffer_.length();
-    ENVOY_CONN_LOG(debug, "TSI: unprotecting buffer size: {}", callbacks_->connection(),
-                   raw_read_buffer_.length());
-    tsi_result status = frame_protector_->unprotect(raw_read_buffer_, buffer);
-    ENVOY_CONN_LOG(debug, "TSI: unprotected buffer left: {} result: {}", callbacks_->connection(),
-                   raw_read_buffer_.length(), tsi_result_to_string(status));
-    result.bytes_processed_ = read_size - raw_read_buffer_.length();
+    return repeatReadAndUnprotect(buffer, result);
   }
 
   ENVOY_CONN_LOG(debug, "TSI: do read result action {} bytes {} end_stream {}",
```

Comment:
No longer covered by tests.

https://storage.googleapis.com/envoy-pr/0357f2f/coverage/source/extensions/transport_sockets/alts/tsi_socket.cc.gcov.html

Reply:
We should not reach this code path any more because handshake_complete_ is always true at this point. Code is updated. PTAL.